### PR TITLE
feat: core deployment tasks

### PR DESCRIPTION
**Pull request by Reinis-FRP**
**Motivation**

Prepare scripts and configuration for OO deployment on https://coredao.org/

**Summary**

Updates scripts for non-rollup chain deployments and adds https://coredao.org network parameters.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [x]  Untested

**Issue(s)**

Fixes https://linear.app/uma/issue/UMA-2042/spec-oo-deployment-to-core-protocol

### CODE REVIEW

File: packages/common/src/hardhat/tasks/collateralWhitelist.ts [hunk 1]
```diff
@@ -10,6 +11,19 @@ import type { CombinedHRE } from "./types";
...
+// LZ chainId mappings from https://github.com/LayerZero-Labs/wrapped-asset-bridge/blob/main/constants/chainIds.json
+const LZ_CHAIN_IDS: Record<number, number> = {
+  1: 101, // ethereum
+  10: 111, // optimism
+  137: 109, // polygon
+  1116: 153, // coredao
+  42161: 110, // arbitrum
+};
+
+const LZ_WRAPPED_TOKEN_BRIDGES: Record<number, string> = {
```
Comment (mrice32) [inline packages/common/src/hardhat/tasks/collateralWhitelist.ts:23]: Interesting: they use LZ as their canonical token bridge?

File: packages/common/src/hardhat/tasks/xchainSetup.ts [hunk 2]
```diff
@@ -525,3 +548,158 @@ task("verify-xchain", "Checks ownership state of cross chain smart contracts")
      * End: Checking L2 State
      *************************************/
   });
+
+task("verify-admin-xchain", "Checks state of L2 smart contracts with admin messenger").setAction(async function (
```
Comment (mrice32) [inline packages/common/src/hardhat/tasks/xchainSetup.ts:552]: This is great! Super useful!

File: packages/core/contracts/cross-chain-oracle/README.md [hunk 3]
```diff
@@ -136,60 +136,117 @@ To finalize **Optimism** transactions see the equivalent script [here](https://g
...
 
-2. Deploy l2 Contracts
+2. Add new network variables in `@uma/common` package:
+
+- Hardhat `defaultConfig.networks` configuration in [packages/common/src/HardhatConfig.ts](/packages/common/src/HardhatConfig.ts). Adding `companionNetworks` parameter pointing to L1 mainnet is required for approved collateral migration task.
+- `etherscan.customChains` configuration in [packages/common/src/HardhatConfig.ts](/packages/common/src/HardhatConfig.ts) if the L2 is not supported by `@nomicfoundation/hardhat-verify` library.
+- `PublicNetworks` configuration in [packages/common/src/PublicNetworks.ts](/packages/common/src/PublicNetworks.ts)
+- `averageBlockTimeSeconds` in [packages/common/src/TimeUtils.ts](/packages/common/src/TimeUtils.ts)
+- Default RPC to `getNodeUrl` in [packages/common/src/ProviderUtils.ts](/packages/common/src/ProviderUtils.ts)
+- Rebuild `@uma/common` package:
```
Comment (mrice32) [inline packages/core/contracts/cross-chain-oracle/README.md:156]: +1 this is great

**Review by mrice32 — APPROVED** [review]
LGTM! This is great @Reinis-FRP

File: packages/common/src/hardhat/tasks/collateralWhitelist.ts (hunk 1 above)
Comment (Reinis-FRP) [reply packages/common/src/hardhat/tasks/collateralWhitelist.ts:23]: yes, The Core Bridge is powered by LayerZero: https://docs.coredao.org/developer/develop-on-core/bridging-with-core/core-bridge-resources

**Pull request merged**